devimint/src/main.rs

```diff
@@ -1114,58 +1114,106 @@ async fn setup(arg: CommonArgs) -> Result<(ProcessManager, TaskGroup)> {
     write_overwrite_async(globals.FM_TEST_DIR.join("env"), env_string).await?;
     info!("Test setup in {:?}", globals.FM_DATA_DIR);
     let process_mgr = ProcessManager::new(globals);
-    let task_group = TaskGroup::new();
-    task_group.install_kill_handler();
-    Ok((process_mgr, task_group))
+    Ok(process_mgr)
+}
+
+async fn cleanup_on_exit(
+    main_process: impl futures::Future<Output = Result<()>>,
+    task_group: TaskGroup,
+    process_mgr: ProcessManager,
+) -> anyhow::Result<()> {
+    // This select makes it possible to exit earlier if a signal is received before
+    // the main process is finished
+    tokio::select! {
+        _ = task_group.make_handle().make_shutdown_rx().await => {
+            info!("Received shutdown signal before finishing setup");
```

Comment:
not sure about "Setup". Should be "main process"?
same for rest of logs in this function.